### PR TITLE
Roy/pm 2095

### PR DESCRIPTION
**Pull request by RoyPollock**
PM-2187, don't use the AS limit, it's just to fragile given how large an AS it inherits from the parent process. Instead, monitor the oom_score in proc which is based on a percentage of available memory used. Do this on a 1 sec timer to have time to print a message and abort.  This still leaves a corefile.

Tested that two ROS webrtc connections can co-exist (there usually is plenty of headroom below the 40% kill threshold)

### CODE REVIEW

File: src/cpp/main.cpp [hunk 1]
```diff
@@ -55,6 +56,36 @@ struct Reap {
...
+                proc_file = fopen(proc_path,"r");
+            }
+            rewind(proc_file);
+            fflush(proc_file);
+            int oom_score;
+            fscanf(proc_file,"%d",&oom_score);
+            if(oom_score > 400) {
+                ROS_INFO(
+                    "Memory went over limit (%d percent)",oom_score/10
+                );
```
Comment (po1) [inline src/cpp/main.cpp]: this should be a `ROS_FATAL`

File: src/cpp/main.cpp [hunk 2]
```diff
@@ -132,6 +158,12 @@ int main(int argc, char **argv) {
         reap_timer.start();
     }
 
+    MemMonitor monitor(NULL);
```
Comment (po1) [inline src/cpp/main.cpp]: why the `void*`?

File: src/cpp/main.cpp [hunk 3]
```diff
@@ -55,6 +56,36 @@ struct Reap {
 
 };
 
+struct MemMonitor {
+
+    MemMonitor(void*) {}
```
Comment (po1) [inline src/cpp/main.cpp]: you don't seem to be using the parameter, is this just to disable the default constructor?

File: src/cpp/main.cpp (hunk 2 above)
Comment (RoyPollock) [inline src/cpp/main.cpp]: Because otherwise I got a template error...

**Pull request merged**